packages/elements/src/photon-client/index.tsx

```diff
@@ -162,24 +162,19 @@ customElement(
     });
 
     createEffect(() => {
-      if (!store()?.authentication.state.isLoading && props.externalUserId != null) {
-        if (
-          (store().authentication.state.user as User).sub?.split('|').reverse()[0] !==
-          props.externalUserId
-        ) {
-          store()?.authentication.logout();
+      if (!store()?.authentication.state.isLoading) {
+        if (!store()?.authentication.state.isAuthenticated && props.autoLogin) {
           const args: any = { appState: {} };
           if (props.redirectPath) {
             args.appState.returnTo = props.redirectPath;
           }
           store()?.authentication.login(args);
-        }
-      }
-    });
-
-    createEffect(() => {
-      if (!store()?.authentication.state.isLoading) {
-        if (!store()?.authentication.state.isAuthenticated && props.autoLogin) {
+        } else if (
+          props.externalUserId != null &&
+          (store().authentication.state.user as User | undefined)?.sub?.split('|').reverse()[0] !==
```

Comment:
I still think we should be doing a substring check instead of doing the whole split thing

Reply:
that still doesnt work because ids arent guaranteed to not be substrings of each other